src/pds_doi_service/core/util/general_util.py

```diff
@@ -13,11 +13,38 @@
 General utility functions for things like logging.
 """
 
+import re
 import logging
 
 from pds_doi_service.core.util.config_parser import DOIConfigUtil
 
 
+def sanitize_json_string(string):
+    """
+    Cleans up extraneous whitespace from the provided string so it may be
+    written to a JSON file. Extraneous whitespace include any before or after
+    the provided string, as well as between words.
+
+    Parameters
+    ----------
+    string : str
+        The string to sanitize.
+
+    Returns
+    -------
+    string : str
+        The provided string, sanitized of extraneous whitespace.
+
+    """
+    # Clean up whitespace between words (including line breaks)
+    string = " ".join(re.split("\\s+", string, flags=re.UNICODE))
```

Comment:
Maybe use `r"\s+"` just to emphasize regexp strings? I dunno 🤷‍♀️ maybe a matter of taste 😋

But rather than `re.split` + `str.join`, I think the `re` module has a function that does this in one step: `re.sub`:

```python
import re
x = '''hello world  how

are

you              doing?'''
y = re.sub(r'\s+', ' ', x)
print(y)
```
outputs `'hello world how are you doing?'`

Reply:
You're absolutely correct, this is a better approach, will add soon